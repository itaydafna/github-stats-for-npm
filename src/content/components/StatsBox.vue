<template>
  <div id="stats-box-container">
    <div id="stats-box">
      <h2 
        v-if="!!repoData.name" 
        class="title">{{ repoData.name }}</h2>
      <table>
        <tbody>
          <tr v-if="!!repoData.stargazers_count">
            <td class="param-td">
              URL:
            </td>
            <td class="value-td">
              <a 
                :href="repoData.html_url" 
                target="_blank">{{ repoData.html_url }}</a>
            </td>
          </tr>
          <tr v-if="!!repoData.stargazers_count">
            <td class="param-td">
              Stars:
            </td>
            <td class="value-td">
              <IconedCounter 
                :counter-val="repoData.stargazers_count">
                <Octicon name="star"/>
              </IconedCounter>
            </td>
          </tr>
          <tr v-if="!!repoData.subscribers_count">
            <td class="param-td">
              Subscribers:
            </td>
            <td class="value-td">
              <IconedCounter 
                :counter-val="repoData.subscribers_count">
                <Octicon name="eye"/>
              </IconedCounter>
            </td>
          </tr>
          <tr v-if="!!repoData.forks">
            <td class="param-td">
              Forks:
            </td>
            <td class="value-td">
              {{ repoData.forks | addCommas }}
            </td>
          </tr>
          <tr v-if="!!repoData.watchers">
            <td class="param-td">
              Watchers:
            </td>
            <td class="value-td">
              {{ repoData.watchers | addCommas }}
            </td>
          </tr>
          <tr v-if="!!repoData.size">
            <td class="param-td">
              Repo Size:
            </td>
            <td class="value-td">
              {{ repoData.size | addCommas }} KB
            </td>
          </tr>
          <tr v-if="repoData.has_issues">
            <td>
              Open Issues:
            </td>
            <td class="value-td">
              {{ repoData.open_issues_count | addCommas }}
            </td>
          </tr>
          <tr v-if="!!repoData.created_at">
            <td class="param-td">
              Created:
            </td>
            <td class="value-td">
              {{ repoData.created_at | formatDate }}
            </td>
          </tr>
          <tr v-if="!!repoData.updated_at">
            <td class="param-td">
              Last Update:
            </td>
            <td class="value-td">
              {{ repoData.updated_at | formatDate }}
            </td>
          </tr>
          <tr v-if="!!repoData.homepage">
            <td class="param-td">
              Homepage:
            </td>
            <td class="value-td">
              <a 
                :href="repoData.homepage" 
                target="_blank">
                {{ repoData.homepage }}
              </a>
            </td>
          </tr>
          <tr v-if="!!repoData.language">
            <td class="param-td">
              Language:
            </td>
            <td class="value-td">
              {{ repoData.language }}
            </td>
          </tr>
          <tr v-if="!!repoData.license && !!repoData.license.name">
            <td class="param-td">
              license:
            </td>
            <td class="value-td">
              {{ repoData.license.name }}
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</template>

<script>
import IconedCounter from './IconedCounter';
import Octicon from 'vue-octicon/components/Octicon.vue';
import 'vue-octicon/icons/star';
import 'vue-octicon/icons/eye';

export default {
  components: {
    IconedCounter,
    Octicon,
  },
  props: { repoData: { type: Object, isRequired: true } },
};
</script>

<style scoped>
#stats-box-container {
  position: relative;
}
#stats-box {
  position: absolute;
  background: #ffffff;
  z-index: 300;
  padding: 20px;
  border-radius: 2%;
  box-shadow: 3px 3px 15px black;
  right: 20px;
  width: 500px;
}
tr {
  line-height: 25px;
  background: #ffffff;
}
.title {
  padding-top: 0;
}
.param-td {
  width: 120px;
}
.value-td {
  min-width: 100%;
  display: flex;
  flex-wrap: wrap;
  padding-left: 10px;
  font-weight: 550;
  word-break: break-all;
}

@media (max-width: 620px) {
  #stats-box-container {
    position: static;
  }
  #stats-box {
    width: 100%;
    right: 0;
    left: 0;
  }
}
</style>
