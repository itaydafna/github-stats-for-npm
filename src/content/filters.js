import Vue from 'vue';

export const addCommas = value => {
  if (!value) return '';
  return value.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
};

export const formatDate = value => {
  try {
    if (!value) return '';
    const d = new Date(value);
    if (isNaN(d)) return '';
    return d
      .toDateString()
      .split(' ')
      .slice(1)
      .join(' ');
  } catch (e) {
    return '';
  }
};
