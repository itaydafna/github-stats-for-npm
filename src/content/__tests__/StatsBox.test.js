import Vue from 'vue';
import { mount } from '@vue/test-utils';
import StatsBox from '../components/StatsBox';
import stats from './mock-stats.json';
import * as filters from '../filters';

describe('StatsBox', () => {
  let wrapper;
  beforeEach(() => {
    Object.keys(filters).forEach(filter => {
      Vue.filter(filter, filters[filter]);
    });
    wrapper = mount(StatsBox, { propsData: { repoData: stats } });
  });
  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
  it('renders stats box', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
