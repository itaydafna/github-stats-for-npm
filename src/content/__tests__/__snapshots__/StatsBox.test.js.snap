// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsBox renders stats box 1`] = `
<div
  id="stats-box-container"
>
  <div
    id="stats-box"
  >
    <h2
      class="title"
    >
      vue-test-utils
    </h2>
     
    <table>
      <tbody>
        <tr>
          <td
            class="param-td"
          >
            
            URL:
          
          </td>
           
          <td
            class="value-td"
          >
            <a
              href="https://github.com/vuejs/vue-test-utils"
              target="_blank"
            >
              https://github.com/vuejs/vue-test-utils
            </a>
          </td>
        </tr>
         
        <tr>
          <td
            class="param-td"
          >
            
            Stars:
          
          </td>
           
          <td
            class="value-td"
          >
            <div>
              <svg
                class="octicon"
                height="16"
                role="presentation"
                version="1.1"
                viewBox="0 0 14 16"
                width="14"
              >
                <path
                  d="M14 6l-4.9-.64L7 1 4.9 5.36 0 6l3.6 3.26L2.67 14 7 11.67 11.33 14l-.93-4.74z"
                />
              </svg>
               
              <span>
                1,883
              </span>
            </div>
          </td>
        </tr>
         
        <tr>
          <td
            class="param-td"
          >
            
            Subscribers:
          
          </td>
           
          <td
            class="value-td"
          >
            <div>
              <svg
                class="octicon"
                height="16"
                role="presentation"
                version="1.1"
                viewBox="0 0 16 16"
                width="16"
              >
                <path
                  d="M8.06 2C3 2 0 8 0 8s3 6 8.06 6C13 14 16 8 16 8s-3-6-7.94-6zM8 12c-2.2 0-4-1.78-4-4 0-2.2 1.8-4 4-4 2.22 0 4 1.8 4 4 0 2.22-1.78 4-4 4zm2-4c0 1.11-.89 2-2 2-1.11 0-2-.89-2-2 0-1.11.89-2 2-2 1.11 0 2 .89 2 2z"
                />
              </svg>
               
              <span>
                69
              </span>
            </div>
          </td>
        </tr>
         
        <tr>
          <td
            class="param-td"
          >
            
            Forks:
          
          </td>
           
          <td
            class="value-td"
          >
            
            230
          
          </td>
        </tr>
         
        <tr>
          <td
            class="param-td"
          >
            
            Watchers:
          
          </td>
           
          <td
            class="value-td"
          >
            
            1,883
          
          </td>
        </tr>
         
        <tr>
          <td
            class="param-td"
          >
            
            Repo Size:
          
          </td>
           
          <td
            class="value-td"
          >
            
            3,802 KB
          
          </td>
        </tr>
         
        <tr>
          <td>
            
            Open Issues:
          
          </td>
           
          <td
            class="value-td"
          >
            
            44
          
          </td>
        </tr>
         
        <tr>
          <td
            class="param-td"
          >
            
            Created:
          
          </td>
           
          <td
            class="value-td"
          >
            
            May 28 2017
          
          </td>
        </tr>
         
        <tr>
          <td
            class="param-td"
          >
            
            Last Update:
          
          </td>
           
          <td
            class="value-td"
          >
            
            Sep 15 2018
          
          </td>
        </tr>
         
        <tr>
          <td
            class="param-td"
          >
            
            Homepage:
          
          </td>
           
          <td
            class="value-td"
          >
            <a
              href="https://vue-test-utils.vuejs.org"
              target="_blank"
            >
              
              https://vue-test-utils.vuejs.org
            
            </a>
          </td>
        </tr>
         
        <tr>
          <td
            class="param-td"
          >
            
            Language:
          
          </td>
           
          <td
            class="value-td"
          >
            
            JavaScript
          
          </td>
        </tr>
         
        <tr>
          <td
            class="param-td"
          >
            
            license:
          
          </td>
           
          <td
            class="value-td"
          >
            
            MIT License
          
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
